libs/expo/betterangels/src/lib/screens/AddEditClient/index.tsx

```diff
@@ -128,39 +129,40 @@ export default function AddEditClient({ id }: { id?: string }) {
   const onSubmit: SubmitHandler<
     UpdateClientProfileInput | CreateClientProfileInput
   > = async (values) => {
-    if (values.contacts && values.contacts?.length > 0) {
-      values.contacts = values.contacts.map((contact) => ({
-        ...contact,
-        phoneNumber: contact.phoneNumber === '' ? null : contact.phoneNumber,
-      }));
-    }
-
-    const filteredSocialMediaProfiles =
-      values.socialMediaProfiles?.filter((item) => item.platformUserId) || [];
+    if (isCreating || isUpdating) return;
+    try {
```

Comment:
I don't think we should be moving all these operations into the `try` block. 

The `try`/`catch` are here because it's possible for something to go wrong when the frontend tries to connect to the backend. So the frontend needs to be protected in case the mutation fails or returns an error. But everything before `let operationResult;` is just the frontend cleaning the data before sending it to the backend, so it should be safe. 

Technically, it is possible for something to go wrong here, but that should be caught/prevented with type checking, field validation, etc. By the time we're cleaning the data here, we know exactly what shape it's going to be, so we don't have to worry about something breaking